### DISCUSSION TITLE
How to set nvim as opener for files? (windows)

### DISCUSSION THREAD

**Discussion by nik-rev** [sxyazi/yazi#1600] [1. Q&A] [answered]
I tried this in `yazi.toml`:

```yaml
[opener]
edit = [
  { run = "nvim %*", for = "windows" },
]
```

But it seems to be running it in the background instead of opening it with neovim

I get this in the right corner:

![image](https://github.com/user-attachments/assets/af2cf7d9-0ca7-44fe-9180-a4f745bd7944)

Also yazi becomes very very slow when i click "enter" on a file. probably because the process in the background

also tried

```yaml
  { run = "nvim %*", for = "windows" },
```

But it didn't change anything

I thought maybe yazi needs `EDITOR` env variable. So i did
```cmd
setx EDITOR nvim
```

To no avail

**Comment by AnirudhG07** [comment]
Can you send your `yazi --debug`? It might be helpful to track issue

**Comment by vbd** [comment]
I'm using Vim instead nvim but my yazi.toml contains the following:
	{ run = 'vim.exe %*',    orphan = true, desc = "vim",           for = "windows" },
- Env for VISUAL  is set to vim.exe
- Env for YAZI_FILE_ONE is set to c:\Program Files\Git\usr\bin\file.exe

**Comment by sxyazi** [comment, marked as answer]
Does `vim` in your `%PATH%` [env variables](https://www.computerhope.com/issues/ch000549.htm#windows11)? Also you should use `block = true` rather than `orphan = true`, https://yazi-rs.github.io/docs/configuration/yazi#opener

**Reply by vbd** [reply]
I think for my workflow block and orphan won't change much. But I will try block = true.
The path to vim.exe is also in env variables.

**Reply by vbd** [reply]
Switched to from orphan to block.
Should have mentioned that my setup with vim.exe works... sorry for confusion.

**Reply by nik-rev** [reply]
> Does `vim` in your `%PATH%` [env variables](https://www.computerhope.com/issues/ch000549.htm#windows11)? Also you should use `block = true` rather than `orphan = true`, https://yazi-rs.github.io/docs/configuration/yazi#opener

I set `block = true` and it works for me now. 
```
	{ run = 'nvim %*',    block = true, desc = "nvim",           for = "windows" },

```

thank you!
